python/source/common/parameter_container.cpp

```diff
@@ -101,7 +103,7 @@ py::object ParameterContainer::get_value() {
   }
 }
 
-ParameterContainer parameter_interface_ptr_to_container(const std::shared_ptr<ParameterInterface>& parameter) {
+ParameterContainer interface_ptr_to_container(const std::shared_ptr<ParameterInterface>& parameter) {
   switch (parameter->get_type()) {
     case StateType::PARAMETER_INT: {
       auto param = std::static_pointer_cast<Parameter<int>>(parameter);
```

Comment:
You should leverage the new accessor method of ParameterInterface with dynamic casting
```suggestion
      auto param = parameter->get_parameter<int>();
```

Or, better yet, you can do it all in one line and skip the intermediate variable:
```
return ParameterContainer(parameter->get_name(), py::cast(parameter->get_parameter_value<int>()), parameter->get_type());
```